### PR TITLE
Fix bug with EphemeralPurger

### PR DESCRIPTION
**Pull request by joshblum**
cc @mmaxim

### CODE REVIEW

**Comment by mmaxim** [comment]
Is it just not running? How did tests not find it?

**Comment by joshblum** [comment]
It's not incorrect but just weird that we immediately `Stop`/`Reset` the timer after setting it and wasn't intended

**Comment by joshblum** [comment]
thinking about it a bit more i think it's safer to use `looping` flag to trigger things so we never have to worry about the loop running and us setting the timer `nil`

**Comment by mmaxim** [comment]
Does that variable to need be protected by a lock?

**Comment by joshblum** [comment]
Ran into some deadlocking issues when trying to exit the loop. to address the changes are:

- added `queueLock` for modifications to the queue so `Start`/`Stop` won't deadlock while holding the main lock and the loop tries to fire (deadlocking while `Start`/`Stop` wait for the `stopCh` to receive. 
- hold the main lock while reading/writing `looping`

**Comment by joshblum** [comment]
reran 50x with sleeps around `Start`/`Stop` w/o issue

File: go/chat/ephemeral_purger.go [hunk 1]
```diff
@@ -158,21 +164,29 @@ func (b *BackgroundEphemeralPurger) Start(ctx context.Context, uid gregor1.UID)
 			storage.NewOutbox(b.G(), b.uid).EphemeralPurge(context.Background())
```
Comment (mmaxim) [inline go/chat/ephemeral_purger.go]: It's a little sloppy that this loop doesn't get shutdown.

Comment (joshblum) [reply go/chat/ephemeral_purger.go]: done!

**Pull request merged**